components/homme/src/test_src/held_suarez_mod.F90

```diff
@@ -252,9 +272,22 @@ subroutine hs0_init_state(elem, hybrid, hvcoord,nets,nete,Tinit)
        enddo
 #endif
 
-       elem(ie)%state%v(:,:,:,:,n0) =0.0D0
-       elem(ie)%state%v(:,:,:,:,nm1)=elem(ie)%state%v(:,:,:,:,n0)
-       elem(ie)%state%v(:,:,:,:,np1)=elem(ie)%state%v(:,:,:,:,n0)
+
+       ! initialize surface pressure to be consistent with topo
+       ! use 11.3pa per meter 
+       elem(ie)%state%ps_v(:,:,n0) = elem(ie)%state%ps_v(:,:,n0) - &
+            11.3*elem(ie)%state%phis(:,:)/g
```

Comment:
For const T it seems that analytical formula is p(z)=C exp(-g z / R_dry T) for dry air and C obtained from the top pressure...